config/initializers/extensions.rb

```diff
@@ -0,0 +1,8 @@
+class Array
+  # Use one_and_only instead of first when dealing with arrays that
+  # should only ever have one element. 
+  def one_and_only
+    raise "Not an array with a single element" unless size == 1
+    first
+  end
```

Comment:
We don't want frontend to return `500` errors if we could manage to return something reasonable.  If there were 2 items in the links field here, we could still serve the page (but might end up highlighting the wrong column, or similar).

I do think making our code clear that we don't expect the array to have more than one entry is good, though.  How about making it log an error to errbit in this case, but still return the first value in the array?  That way, we get reports in errbit, but the users still get their content. We've used that pattern in the past a few times, and it's been useful.

I think we should also make the schemas validate that there is at most one item in these arrays, too.